server/interpreter.js

```diff
@@ -871,41 +866,62 @@ Interpreter.prototype.initFunction_ = function() {
   var wrapper;
   var identifierRegexp = /^[A-Za-z_$][\w$]*$/;
   // Function constructor.
-  wrapper = function(var_args) {
-    if (arguments.length) {
-      var code = String(arguments[arguments.length - 1]);
-    } else {
-      var code = '';
-    }
-    var argsStr = Array.prototype.slice.call(arguments, 0, -1).join(',').trim();
-    if (argsStr) {
-      var args = argsStr.split(/\s*,\s*/);
-      for (var i = 0; i < args.length; i++) {
-        var name = args[i];
-        if (!identifierRegexp.test(name)) {
-          intrp.throwError(intrp.SYNTAX_ERROR,
-              'Invalid function argument: ' + name);
-        }
+  new this.NativeFunction({
+    id: 'Function', length: 1,
```

Comment:
Split this line.  I'd be ok with x and y being on the same line, but id and length are unrelated.
Here and several places below.

Reply:
Thy aren't that unrelated; they are in effect the function signature: `id: Foo, length: 3` means `Foo(arg0, arg1, arg2)`, but without naming the arguments.